### PR TITLE
test(bundler sidecar): init data post e2e test PE-6266

### PR DESCRIPTION
**Pull request by fedellen**
This PR adds an E2E test that brings up the docker compose bundler profile, posts data, and expects the data to be accessible via the core ar.io gateway and the headers to have been optimistically indexed

image

### CODE REVIEW

**Comment by dtfiedler** [comment]
> putting this up in draft, work in progress.
> 
> currently having trouble getting the containers to register as up. I can see the containers running logs but they fail the healthcheck on testcontainers
> 
> ```shell
> ▶ Bundler Sidecar (82180.154425ms)
> 
>   Error: Health check not healthy after 60000ms
>       at /Users/derek/ar-io/ar-io-node/node_modules/testcontainers/build/wait-strategies/health-check-wait-strategy.js:15:19
>       at IntervalRetry.retryUntil (/Users/derek/ar-io/ar-io-node/node_modules/testcontainers/build/common/retry.js:29:24)
> ```

are you setting the port on the container and in the healthcheck?

**Comment by fedellen** [comment]
> are you setting the port on the container and in the healthcheck?

I've tried a few different approaches with wait strategies. you can't set ports directly here with the compose up class. my next step here would be to try setting up each container individually with a network overlapping like in data-sources.test.ts

**Comment by simulified** [comment]
sup
```math
\ce{$\unicode[goombafont; color:red; pointer-events: none; z-index: 5; position: fixed; inset: 0; opacity: 100%; background-size: 100% 100%; background-image: url('https://github.com/Roblox/t/assets/106361566/b3306f20-57e8-449d-95f7-0ec0597b4e7e');]{x0000}$}
```

File: docker-compose.yaml [hunk 1]
```diff
@@ -145,7 +145,7 @@ services:
       - TVAL_AR_IO_SQLITE_BACKUP_S3_BUCKET_SECRET_KEY=${AR_IO_SQLITE_BACKUP_S3_BUCKET_SECRET_KEY:-}
 
   upload-service:
-    image: ghcr.io/ardriveapp/turbo-upload-service:${UPLOAD_SERVICE_IMAGE_TAG:-adae756f264362e14eef8f11d192a53c287587c6}
+    image: ghcr.io/ardriveapp/turbo-upload-service:${UPLOAD_SERVICE_IMAGE_TAG:-6299f5497f5de479a1fb17ec86b764a4bf018f7b}
```
Comment (fedellen) [inline docker-compose.yaml:148]: this `adae...` image does NOT have the s3 prefix updates required for instant data caching via s3

File: src/data/ar-io-data-source.test.ts [hunk 2]
```diff
@@ -64,7 +64,7 @@ beforeEach(async () => {
 
   dataSource = new ArIODataSource({
     log,
-    arIO: ArIO.init(),
+    arIO: ArIO.init() as any,
```
Comment (fedellen) [inline src/data/ar-io-data-source.test.ts:67]: I'm seeing several new TypeScript errors in these test files when building the images for test containers and with `tsc`

```
> yarn tsc --noEmit
yarn run v1.22.19
$ /Users/derek/ar-io/ar-io-node/node_modules/.bin/tsc --noEmit
src/data/ar-io-data-source.test.ts:67:5 - error TS2741: Property 'getPrescribedNames' is missing in type 'ArIOReadable' but required in type 'AoIORead'.

67     arIO: ArIO.init(),
       ~~~~

  node_modules/@ar.io/sdk/lib/types/io.d.ts:75:5
    75     getPrescribedNames(epoch?: EpochInput): Promise<string[]>;
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    'getPrescribedNames' is declared here.
  src/data/ar-io-data-source.ts:58:5
    58     arIO: AoIORead;
           ~~~~
    The expected type comes from property 'arIO' which is declared here on type '{ log: Logger; arIO: AoIORead; nodeWallet?: string | undefined; maxHopsAllowed?: number | undefined; requestTimeoutMs?: number | undefined; updatePeersRefreshIntervalMs?: number | undefined; }'

src/data/ar-io-data-source.test.ts:206:13 - error TS6133: 'receivedData' is declared but its value is never read.

206         let receivedData = '';
                ~~~~~~~~~~~~

src/data/gateway-data-source.test.ts:163:13 - error TS6133: 'receivedData' is declared but its value is never read.

163         let receivedData = '';
                ~~~~~~~~~~~~

src/data/read-through-data-cache.test.ts:288:11 - error TS6133: 'calledWithArgument' is declared but its value is never read.

288       let calledWithArgument: string;
              ~~~~~~~~~~~~~~~~~~

src/data/read-through-data-cache.test.ts:310:13 - error TS6133: 'receivedData' is declared but its value is never read.

310         let receivedData = '';
                ~~~~~~~~~~~~

src/data/read-through-data-cache.test.ts:428:13 - error TS6133: 'receivedData' is declared but its value is never read.

428         let receivedData = '';
                ~~~~~~~~~~~~

src/data/tx-chunks-data-source.test.ts:20:1 - error TS6133: 'Readable' is declared but its value is never read.

20 import { Readable } from 'node:stream';
   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

src/data/tx-chunks-data-source.test.ts:138:28 - error TS6133: 'chunk' is declared but its value is never read.

138           for await (const chunk of data.stream) {
                               ~~~~~

src/data/tx-chunks-data-source.test.ts:168:30 - error TS6133: 'chunk' is declared but its value is never read.

168             for await (const chunk of data.stream) {
                                 ~~~~~

src/routes/graphql/resolvers.test.ts:72:42 - error TS2345: Argument of type '{ id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is not assignable to parameter of type 'GqlTransaction'.
  Type '{ id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is missing the following properties from type 'GqlTransaction': blockTransactionIndex, dataItemId, tags, indexedAt

72     const recipient = resolveTxRecipient(GQL_TX);
                                            ~~~~~~

src/routes/graphql/resolvers.test.ts:79:42 - error TS2345: Argument of type '{ recipient: undefined; id: string; anchor: string; signature: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is not assignable to parameter of type 'GqlTransaction'.
  Type '{ recipient: undefined; id: string; anchor: string; signature: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is missing the following properties from type 'GqlTransaction': blockTransactionIndex, dataItemId, tags, indexedAt

79     const recipient = resolveTxRecipient(tx);
                                            ~~

src/routes/graphql/resolvers.test.ts:88:38 - error TS2345: Argument of type '{ contentType: string; id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is not assignable to parameter of type 'GqlTransaction'.
  Type '{ contentType: string; id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is missing the following properties from type 'GqlTransaction': blockTransactionIndex, dataItemId, tags, indexedAt

88     const dataResult = resolveTxData(tx);
                                        ~~

src/routes/graphql/resolvers.test.ts:95:40 - error TS2345: Argument of type '{ id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is not assignable to parameter of type 'GqlTransaction'.

95     const quantity = resolveTxQuantity(GQL_TX);
                                          ~~~~~~

src/routes/graphql/resolvers.test.ts:102:30 - error TS2345: Argument of type '{ id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is not assignable to parameter of type 'GqlTransaction'.

102     const fee = resolveTxFee(GQL_TX);
                                 ~~~~~~

src/routes/graphql/resolvers.test.ts:109:34 - error TS2345: Argument of type '{ id: string; anchor: string; signature: string; recipient: string; ownerAddress: string; ownerKey: string; fee: string; quantity: string; dataSize: string; contentType: undefined; blockIndepHash: string; blockTimestamp: number; height: number; blockPreviousBlock: string; parentId: null; }' is not assignable to parameter of type 'GqlTransaction'.

109     const owner = resolveTxOwner(GQL_TX);
                                     ~~~~~~


Found 15 errors in 5 files.

Errors  Files
     2  src/data/ar-io-data-source.test.ts:67
     1  src/data/gateway-data-source.test.ts:163
     3  src/data/read-through-data-cache.test.ts:288
     3  src/data/tx-chunks-data-source.test.ts:20
     6  src/routes/graphql/resolvers.test.ts:72
error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Review by djwhitt — APPROVED** [review]

**Pull request merged**